Grbl_Esp32/src/Spindles/VFDSpindle.cpp

```diff
@@ -0,0 +1,456 @@
+                        }
+                    case 3:
+                        parser  = instance->get_status_ok(next_cmd);
+                        pollidx = 1;
+
+                        // we could complete this in case parser == nullptr with some ifs, but let's
+                        // just keep it easy and wait an iteration.
+                        break;
+                }
+
+                // If we have no parser, that means get_status_ok is not implemented (and we have
+                // nothing resting in our queue). Let's fall back on a simple continue.
+                if (parser == nullptr) {
+                    vTaskDelay(VFD_RS485_POLL_RATE);
+                    continue;  // main while loop
+                }
+            }
+
+            {
+                // Grabbed the command. Add the CRC16 checksum:
+                auto crc16 = ModRTU_CRC(next_cmd.msg, next_cmd.tx_length);
+
+                next_cmd.tx_length += 2;
+                next_cmd.rx_length += 2;
+
+                // add the calculated Crc to the message
+                next_cmd.msg[next_cmd.tx_length - 1] = (crc16 & 0xFF00) >> 8;
+                next_cmd.msg[next_cmd.tx_length - 2] = (crc16 & 0xFF);
+
+#ifdef VFD_DEBUG_MODE
+                if (parser == nullptr) {
+                    report_hex_msg(next_cmd.msg, "RS485 Tx: ", next_cmd.tx_length);
+                }
+#endif
+            }
+
+            // Assume for the worst, and retry...
+            int retry_count = 0;
+            for (; retry_count < MAX_RETRIES; ++retry_count) {
+                // Flush the UART and write the data:
+                uart_flush(VFD_RS485_UART_PORT);
+                uart_write_bytes(VFD_RS485_UART_PORT, reinterpret_cast<const char*>(next_cmd.msg), next_cmd.tx_length);
+
+                // Read the response
+                uint16_t read_length = uart_read_bytes(VFD_RS485_UART_PORT, rx_message, next_cmd.rx_length, RESPONSE_WAIT_TICKS);
+
+                // Generate crc16 for the response:
+                auto crc16response = ModRTU_CRC(rx_message, next_cmd.rx_length - 2);
+
+                if (read_length == next_cmd.rx_length &&                             // check expected length
+                    rx_message[0] == VFD_RS485_ADDR &&                               // check address
+                    rx_message[read_length - 1] == (crc16response & 0xFF00) >> 8 &&  // check CRC byte 1
+                    rx_message[read_length - 2] == (crc16response & 0xFF)) {         // check CRC byte 1
+
+                    // success
+                    unresponsive = false;
+                    retry_count  = MAX_RETRIES + 1;  // stop retry'ing
+
+                    // Should we parse this?
+                    if (parser != nullptr && !parser(rx_message, instance)) {
+#ifdef VFD_DEBUG_MODE
+                        report_hex_msg(next_cmd.msg, "RS485 Tx: ", next_cmd.tx_length);
+                        report_hex_msg(rx_message, "RS485 Rx: ", read_length);
+
```

Comment:
Extra blank line?

Reply:
Should be below the #endif to make it more readable. Fixed now.